lib/seafoam/graph.rb

```diff
@@ -87,6 +89,18 @@ def id_and_label
     def inspect
       "<Node #{id} #{node_class}>"
     end
+
+    def visit(&block)
+      block.call(self)
+    end
```

Comment:
This could be inlined, but it also adds some value for readability so seems fine.

Reply:
Right. I only added it to use somewhat standard naming for the visitor pattern.